### PR TITLE
DCMAW-9802: Update Error Result type

### PR DESCRIPTION
**Pull request by sandy-may**
​DCMAW-9802
### What changed
Add an errorCategory to the Result type
Does not mock out request processing in any unit tests

### Why did it change
Enriching the logs
Tests for validating the incoming request should be handled in the lambda to enable unit integration.

## Checklists

- [x] There is a ticket raised for this PR that is present in the branch name
- [x] No PII data logged. [See guidance here](https://govukverify.atlassian.net/wiki/spaces/DCMAW/pages/3502407722/PII+Logging+Considerations)
- [x] Demo to a BA, TA, and the team.
- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

### CODE REVIEW

File: backend-api/src/functions/utils/tests/result.test.ts [hunk 1]
```diff
@@ -1,35 +1,36 @@
 import { errorResult, Result, successResult } from "../result";
 
 describe("Result", () => {
-  describe("Given there is an error and the return type is not specified", () => {
-    it("An error response is returned with a string", () => {
+  describe("Given there is an errord", () => {
```
Comment (jamesamooney) [inline backend-api/src/functions/utils/tests/result.test.ts:4]:
```suggestion
  describe("Given there is an error", () => {
```

**Review by jamesamooney — APPROVED** [review]

File: backend-api/src/functions/asyncToken/requestService/requestService.ts [hunk 2]
```diff
@@ -1,61 +1,80 @@
...
-    request: APIGatewayProxyEvent,
-  ) => Result<IDecodedClientSecrets>;
-}
 
 export interface IDecodedAuthorizationHeader {
   clientId: string;
   clientSecret: string;
 }
+
+interface IProcessRequest {
```
Comment (sandy-may) [inline backend-api/src/functions/asyncToken/requestService/requestService.ts:10]:
```suggestion
interface IRequestService {
```

File: backend-api/src/functions/asyncToken/requestService/requestService.ts [hunk 3]
```diff
@@ -1,61 +1,80 @@
...
   clientSecret: string;
 }
+
+interface IRequestService {
+  getClientCredentials: (
+    headers: APIGatewayProxyEventHeaders,
+  ) => Result<IDecodedAuthorizationHeader>;
+  validateBody: (body: string | null) => Result<null>;
+}
+export const requestService: IProcessRequest = {
```
Comment (sandy-may) [inline backend-api/src/functions/asyncToken/requestService/requestService.ts:16]:
```suggestion
export const requestService: IRequestService = {
```

File: backend-api/src/functions/asyncToken/requestService/requestService.ts [hunk 4]
```diff
@@ -1,61 +1,80 @@
...
+    getClientCredentials(headers),
+  validateBody: (body: string | null) => validateBody(body),
+};
+
+const getClientCredentials = (
+  headers: APIGatewayProxyEventHeaders,
+): Result<IDecodedClientSecrets> => {
+  const authorizationHeader = headers.Authorization;
+  if (!authorizationHeader) {
+    return errorResult({
```
Comment (jamesamooney) [inline backend-api/src/functions/asyncToken/requestService/requestService.ts:27]: As discussed, please can you check we have test coverage of this

Comment (sandy-may) [reply backend-api/src/functions/asyncToken/requestService/requestService.ts:27]: done

**Review by jamesamooney — APPROVED** [review]

**Review by J-son1 — APPROVED** [review]

**Pull request merged**